R/make_recipe.R

```diff
@@ -3,26 +3,98 @@
 #' @description
 #' The `make_recipe()` function is used to quickly apply common data preprocessing techniques
 #'
-#' @param data A dataframe containing training data, validation data, X, and y
-#' @param recipe A string specifying which recipe to apply to the data
-#' @param create_train_test Boolean, if TRUE will partition data into train and test
-#' @param create_train_valid_test Boolean, if TRUE will partition data into train, valid, and test
+#' @param X A dataframe containing training data, validation data, and testing data (should contain X and y).
+#' @param y The name of the response column (as a string, e.g. "response_variable").
+#' @param recipe A string specifying which recipe to apply to the data. See "The recipe parameter" section below for details.
+#' @param splits_to_return A string speciying how to split the data. "train_test" to return train and test splits, "train_test_valid" to return train, test, and validation data, "train" to return all data without splits.
+#' @param random_seed An integer. The random seed to set for splitting data to create reproducible results. By default NULL
+#' @param train_valid_prop A float. The proportion to split the data by. Should range between 0 to 1. By default = 0.8
 #'
-#' @return A list of dataframes e.g. list(train, valid, test)
-#'
-#' @section Recipes
+#' @return A list of dataframes e.g. list(X_train, X_valid, X_test, y_train, y_valid, y_test)
 #'
+#' @section The recipe parameter:
 #' The following recipes are available currently available to pass into the `recipe` parameter:
+#' * "ohe_and_standard_scaler" - Apply one hot encoding to categorical features and standard scaler to numeric features
 #'
-#' "ohe_and_standard_scaler"
+#' More recipes are under development and will be released in future updates.
 #'
-#' Apply one hot encoding to categorical features and standard scaler to numeric features
+#' @md
```

Comment:
Was this added intentionally?

```suggestion
#'
```

Reply:
Yes this was added intentionally. It is telling R that when building the documentation to treat my input as a markdown. It allows for much easier format when writing doc strings. So this must be kept.